.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,8 @@
+<!--
+Thank you for creating a pull request. Before submitting, please note the following:
+
+* If your pull request implements a new feature, please raise an issue to discuss it before sending code. In many cases features are absent for a reason.
+* This message body should clearly illustrate what problems it solves. If there are related issues, remember to reference them.
+* If necessary, and you know how, consider squashing your commits before merging. (This isn't essential but it keeps the history cleaner.)
```

Comment:
This can be done by the person merging using the "Squash & commit" setting for the merge button.